stock-service/src/main/java/rs/banka4/stock_service/domain/options/db/BlackHolesOptionMaker.java

```diff
@@ -0,0 +1,144 @@
+
+        return calculateOptionPrice(S, K, T, r, sigma, type);
+    }
+
+
+    public static String makeOptionTicker(
+        OffsetDateTime expiry,
+        String ticker,
+        OptionType type,
+        int strikePrice
+    ) {
+        String datePart = expiry.format(DateTimeFormatter.ofPattern("yyMMdd"));
+        String optionTypeChar = type == OptionType.CALL ? "C" : "P";
+        String strikeFormatted = String.format("%08d", strikePrice * 100);
+        return ticker.toUpperCase() + datePart + optionTypeChar + strikeFormatted;
+    }
+
+    public List<Option> generateOptions(Stock stock, BigDecimal currentPrice) {
+        List<Option> options = new ArrayList<>();
+        OffsetDateTime now = OffsetDateTime.now(ZoneOffset.UTC);
+
+        int basePrice =
+            currentPrice.setScale(0, BigDecimal.ROUND_HALF_UP)
+                .intValue();
+        List<Integer> strikePrices = new ArrayList<>();
+        for (int i = -5; i <= 5; i++) {
+            strikePrices.add(basePrice + i);
+        }
+
+        List<OffsetDateTime> expiries = new ArrayList<>();
+        for (int i = 6; i <= 30; i += 6) {
+            expiries.add(now.plusDays(i));
+        }
+        for (int i = 60; i <= 210; i += 30) {
+            expiries.add(now.plusDays(i));
+        }
+
+        for (OffsetDateTime expiry : expiries) {
+            for (Integer strike : strikePrices) {
+                MonetaryAmount strikeAmount =
+                    new MonetaryAmount(BigDecimal.valueOf(strike), CurrencyCode.USD);
+
+                String tickerC =
+                    makeOptionTicker(expiry, stock.getTicker(), OptionType.CALL, strike);
+                String tickerP =
+                    makeOptionTicker(expiry, stock.getTicker(), OptionType.PUT, strike);
+
+                int daysToExpiry = (int) ChronoUnit.DAYS.between(now, expiry);
+                int openInterest = Math.max(0, 500 - daysToExpiry * 5);
+
+                Option callOption =
+                    Option.builder()
+                        .stock(stock)
+                        .optionType(OptionType.CALL)
+                        .strikePrice(strikeAmount)
+                        .impliedVolatility(0.4)
+                        .openInterest(openInterest)
+                        .settlementDate(expiry)
+                        .name(stock.getTicker() + "-CALL-" + strike + "-" + expiry.toLocalDate())
+                        .ticker(tickerC)
+                        .build();
+
+                Option putOption =
+                    Option.builder()
+                        .stock(stock)
+                        .optionType(OptionType.PUT)
+                        .strikePrice(strikeAmount)
+                        .impliedVolatility(0.4)
+                        .openInterest(openInterest)
+                        .settlementDate(expiry)
+                        .name(stock.getTicker() + "-PUT-" + strike + "-" + expiry.toLocalDate())
```

Comment:
this isn't the format suggested in the specification - are we using this one?  is it documented somewhere?